core/src/main/java/org/jruby/ext/bigdecimal/RubyBigDecimal.java

```diff
@@ -127,6 +127,12 @@ public class RubyBigDecimal extends RubyNumeric {
     private static final double SQRT_10 = 3.162277660168379332;
     private static final long NEGATIVE_ZERO_LONG_BITS = Double.doubleToLongBits(-0.0);
 
+    private static RubyBigDecimal POSITIVE_ZERO;
+    private static RubyBigDecimal NEGATIVE_ZERO;
+    private static RubyBigDecimal POSITIVE_INFINITY;
+    private static RubyBigDecimal NEGATIVE_INFINITY;
+    private static RubyBigDecimal NAN;
+
```

Comment:
These cannot be static fields because:

  1. people with multiple Ruby runtimes in a single JVM will get values from other runtimes and bad things happen
  2. can potentially leak entire runtime since Runtime pins the entire system

Reply:
@enebo thanks for your review. i updated my PR using setConstant and setConstantVisibility.

Reply:
If we notice a performance issue with constant table lookup here we can promote these to ordinary fields in some per-runtime Java object but this looks fine.